pallets/pool-registry/src/lib.rs

```diff
@@ -387,12 +402,62 @@ pub mod pallet {
 				},
 			);
 
-			Self::deposit_event(Event::MetadataSet {
-				pool_id,
-				metadata: checked_metadata,
-			});
+			Ok(checked_metadata)
+		}
+	}
 
-			Ok(())
+	impl<T: Config> cfg_traits::PoolMetadata<T::Balance, VersionedMultiLocation> for Pallet<T> {
+		type AssetMetadata = orml_asset_registry::AssetMetadata<T::Balance, Self::CustomMetadata>;
+		type CustomMetadata = CustomMetadata;
+		type PoolId = T::PoolId;
+		type PoolMetadata = PoolMetadataOf<T>;
+		type TrancheId = T::TrancheId;
+
+		fn get_pool_metadata(pool_id: Self::PoolId) -> Option<Self::PoolMetadata> {
+			PoolMetadata::<T>::get(pool_id)
+		}
+
+		fn set_pool_metadata(pool_id: Self::PoolId, metadata: Vec<u8>) -> DispatchResult {
+			Self::do_set_metadata(pool_id, metadata).map(|_| ())
+		}
+
+		fn get_tranche_token_metadata(
+			pool_id: Self::PoolId,
+			tranche_id: Self::TrancheId,
+		) -> Option<Self::AssetMetadata> {
+			let currency_id = T::TrancheCurrency::generate(pool_id, tranche_id).into();
+			T::AssetRegistry::metadata(&currency_id)
+		}
```

Comment:
I would opt in for making this not returning `Option<>` but rather throw an error or is there some reasoning to assume the consumer will set the metadata if not found?